invenio_files_rest/models.py

```diff
@@ -433,9 +433,16 @@ def delete(cls, bucket_id):
 
     @ensure_unlocked()
     def remove(self):
-        """Permanently remove a bucket and all objects."""
-        # 1) remove all object versions 2) remove bucket.
-        raise NotImplementedError()
+        """Permanently remove a bucket and all objects (including versions).
+
+        Note the method does not remove the associated file instances which
```

Comment:
``BucketTag.remove`` -> ``ObjectVersion.remove()``?

With:

> Shall we enhance the note with .. note::?

Did you mean:

1)
```
.. note:: 

   Note the method does not remove the associated file instances which 
   must be garbage collected.
```

or:

2)
```
.. note::

   This should only be used when you want to permanently remove a bucket.
```